libraries/disp/viewers/helpers/scalecontrol.h

```diff
@@ -0,0 +1,271 @@
+
+    //=========================================================================================================
+    /**
+     * Initialize the spinbox settings.
+     */
+    void initSpinBox();
+
+    //=========================================================================================================
+    /**
+     * Initialize the slider settings
+     */
+    void initSlider();
+
+    //=========================================================================================================
+    /**
+     * Callback executed whenever the value of the spinbox has changed.
+     * @param dScale
+     */
+    void spinBoxChanged(double dScale);
+
+    //=========================================================================================================
+    /**
+     * Callback executed whenever the position of the slider has changed.
+     * @param dScale
+     */
+    void sliderChanged(int dScale);
+
+    //=========================================================================================================
+    /**
+     * Member method to set the range of the slider. Normally this is set to [1,1000] and there should be no reason
+     * to change it since the non-linear mapping function between the slider and the spinbox should take care of
+     * addapting the actual value of the control (the spinbox) with the values of the slider. Not even if there is
+     * a need to invert the behavior of the slider since the invertSlider should take care of that.
+     * @param [in] iMin Minimum value for the slider.
+     * @param [in] iMax Maximum value for the slider.
+     */
+    void setSliderRange(int min, int max);
+
+    //=========================================================================================================
+    /**
+     * Member function to map the value of the spinbox to a position in the slider bar consistent with the maximum
+     * and minimum levels of the control.
+     * @param [in] dIn Value of the spinbox or the control.
+     * @return [out] int value to set the slider to, consistent with the range of the slider.
+     */
+    inline int mapSpinBoxToSlider(double dIn);
+
+    //=========================================================================================================
+    /**
+     * Member function to map the value of a slider poisition with a value for the control or the spinbox. This
+     * functions adapts the ranges of the control/spinbox to the ranges of the slider for consistency between ranges.
+     * @param [in] iIn Integer value representing the position fo the slider.
+     * @return [out] dOut Double value to set the spinbox to.
+     */
+    inline double mapSliderToSpinBox(int iIn);
+
+    //=========================================================================================================
+    /**
+     * Memberfunction to update the constant values of the mapping functions between spinbox and the slider. These constant

```

Comment:
Memberfunction -> Member function

Might be unnecessary to state this is a member function